File: Producto/backend/src/main/java/backend/com/produccion/application/dto/HCItemOCItemLinkDTO.java
package backend.com.produccion.application.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class HCItemOCItemLinkDTO {

    private Long hcItemId;
    private Long ocItemId;
    private BigDecimal cantidadAsignada;
}

File: Producto/frontend/src/remote/DTO/HCItemOCItemLinkDTO.js
/**
 * DTO de la tabla puente HCItem ↔ OCItem (asignación N:M).
 * Espejo de HCItemOCItemLinkDTO.java del backend.
 */
export class HCItemOCItemLinkDTO {
    constructor(data = {}) {
        this.hcItemId = data.hcItemId ?? null;
        this.ocItemId = data.ocItemId ?? null;
        this.cantidadAsignada = data.cantidadAsignada ?? 0;
    }
}
